homeassistant/core.py

```diff
@@ -408,6 +433,8 @@ def async_fire(self, event_type: str, event_data=None,
         for func in listeners:
```

Comment:
Just use async_run_job? No reason to waste another trip to the event loop if a callback can be run immediately.